internal/handlers/publishers.go

```diff
@@ -3,30 +3,49 @@ package handlers
 import (
 	"github.com/gofiber/fiber/v2"
 	"github.com/italia/developers-italia-api/internal/common"
-	db "github.com/italia/developers-italia-api/internal/database"
 	"github.com/italia/developers-italia-api/internal/models"
 	"github.com/italia/developers-italia-api/internal/requests"
+	"gorm.io/gorm"
 )
 
-func GetPublishers(ctx *fiber.Ctx) error {
+type PublisherInterface interface {
```

Comment:
I think we can define an interface, something like

```golang
type Resource[T] interface {
  GetAll(ctx *fiber.Ctx) error
  Get(ctx *fiber.Ctx) error
  Post(ctx *fiber.Ctx) error
}
```
and implement it with Go 1.18's generics for all resources

```golang
func (m *Resource[T]) Get(ctx *fiber.Ctx) error {
...
}
```

or whatever the syntax is.

WDYT?

Reply:
Yes sounds cool!
Do you want to show me your idea in the next PR? 🤩

Reply:
Sure, let's do it as a future step :+1: